plugins/modules/nd_interface_flow_rules.py

```diff
@@ -0,0 +1,440 @@
+    flow_rule: my_FlowRule
+    nodes:
+        - node_id: 1
+          node_name: my_node_1
+          ports:
+            - eth1/2
+        - node_id: 3
+          node_name: my_node_3
+          ports: []
+    state: present
+
+- name: Update a Physical Interface Flow Rule by removing all nodes
+  cisco.nd.nd_interface_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    flow_rule: my_FlowRule
+    nodes: []
+    state: present
+
+- name: Update a Physical Interface Flow Rule by adding subnet 10.10.1.0/24
+  cisco.nd.nd_interface_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    flow_rule: my_FlowRule
+    subnets:
+      - 10.10.0.0/24
+      - 10.10.1.0/24
+    state: present
+
+- name: Update a Physical Interface Flow Rule by deleting subnet 10.10.0.0/24
+  cisco.nd.nd_interface_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    flow_rule: my_FlowRule
+    subnets:
+      - 10.10.1.0/24
+    state: present
+
+- name: Update a Physical Interface Flow Rule by deleting all subnets
+  cisco.nd.nd_interface_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    flow_rule: my_FlowRule
+    subnets: []
+    state: present
+
+- name: Query a specific Physical Interface Flow Rule
+  cisco.nd.nd_interface_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    flow_rule: my_FlowRule
+    state: query
+
+- name: Query all Physical Interface Flow Rules
+  cisco.nd.nd_interface_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    flow_rule_type: physical
+    state: query
+
+- name: Query all Interface Flow Rules
+  cisco.nd.nd_interface_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    state: query
+
+- name: Delete a Physical Interface Flow Rule
+  cisco.nd.nd_interface_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    flow_rule: my_FlowRule
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible_collections.cisco.nd.plugins.module_utils.ndi import NDI
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec, sanitize_dict
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.constants import INTERFACE_FLOW_RULES_TYPES_MAPPING, INTERFACE_FLOW_RULES_STATUS_MAPPING
+import copy
+
+
+def create_flow_rules_node_port(nodes=None, existing_nodes=None, flow_rule_type=None):
+    nodes_to_add = []
+    nodes_to_update = []
+    if isinstance(nodes, list):
+        existing_nodes_set = {item["nodeId"] for item in existing_nodes}
+        all_nodes_set = sorted(existing_nodes_set.union({item["node_id"] for item in nodes}))
+        for node_id in all_nodes_set:
+            check_node_input_id = any(node_input["node_id"] == node_id for node_input in nodes)
+
+            if check_node_input_id and node_id not in existing_nodes_set:
+                node_input = next((node for node in nodes if node["node_id"] == node_id))
```

Comment:
what happens when node_id is not provided? Think inside nodes we should make arguments required that are needed for lookups.

Reply:
I will make node_id and node_name required for this.